### PR TITLE
Prameterize kmsid and allow updating connectiondetails

### PR DESCRIPTION
**Pull request by shylesh**
Signed-off-by: Shylesh Kumar Mohan <shmohan@redhat.com>

### CODE REVIEW

**Comment by shylesh** [comment]
rebased on latest master

File: ocs_ci/utility/kms.py [hunk 1]
```diff
@@ -774,11 +774,11 @@ def update_csi_kms_vault_connection_details(update_config):
         csi_kms_conf.get()
     except CommandFailed:
         raise KMSConnectionDetailsError(
-            "CSI kms connection details doesn't exists"
-            "can't continue with update"
+            "CSI kms connection details doesn't exists" "can't continue with update"
```
Comment (petr-balogh) [inline ocs_ci/utility/kms.py:777]: I think that this is missing space between `exists` and `can't` and will lead to `existscan't` . So I think that also when black moved this to one line you can omit the middle `" "`.

**Comment by shylesh** [comment]
```
oc -n openshift-storage get ConfigMap csi-kms-connection-details -n openshift-storage -o yaml
14:54:30 - MainThread - ocs_ci.utility.templating - INFO - apiVersion: v1
data:
  1-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "rbd-encryption", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "rbd-encryption",
    "VAULT_TOKEN_NAME": "/home/rgeorge/ocs-ci/ocs_ci/templates/CSI/rbd/csi-kms-secret.yaml",
    "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE": "cert.pem", "VAULT_CLIENT_KEY_FILE":
    "privkey.pem"}'
  10-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "kv-v2", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "", "VAULT_TOKEN_NAME":
    "ocs-kms-token", "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE":
    "cert.pem", "VAULT_CLIENT_KEY_FILE": "privkey.pem"}'
  2-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "rbd-encryption", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "rbd-encryption",
    "VAULT_TOKEN_NAME": "ocs-kms-token", "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE":
    "cert.pem", "VAULT_CLIENT_KEY_FILE": "privkey.pem"}'
  3-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "rbd-encryption", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "rbd-encryption",
    "VAULT_TOKEN_NAME": "ocs-kms-token", "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE":
    "cert.pem", "VAULT_CLIENT_KEY_FILE": "privkey.pem"}'
kind: ConfigMap
metadata:
  creationTimestamp: '2021-07-15T08:06:13Z'
  name: csi-kms-connection-details
  namespace: openshift-storage
  resourceVersion: '1956846'
  uid: b027c685-46f1-450a-a76a-eb9580ea595b

File: ocs_ci/utility/kms.py (hunk 1 above)
Comment (shylesh) [reply ocs_ci/utility/kms.py]: done

**Comment by shylesh** [comment]
rebased on latest master

**Review by rachael-george — DISMISSED** [review]

File: ocs_ci/helpers/helpers.py [hunk 2]
```diff
@@ -505,6 +505,7 @@ def create_storage_class(
     provisioner=None,
     rbd_thick_provision=False,
     encrypted=False,
+    encryptionkmsid=None,
```
Comment (clacroix12) [inline ocs_ci/helpers/helpers.py:508]: Can we make this more readable with underscores? `encryption_kms_id`?

File: ocs_ci/utility/kms.py [hunk 3]
```diff
@@ -752,6 +752,41 @@ def create_vault_csi_kms_connection_details(
...
+        resource_name=constants.VAULT_KMS_CSI_CONNECTION_DETAILS,
+        kind="ConfigMap",
+        namespace=constants.OPENSHIFT_STORAGE_NAMESPACE,
+    )
+
+    try:
+        csi_kms_conf.get()
+    except CommandFailed:
+        raise KMSConnectionDetailsError(
+            "CSI kms connection details doesn't exists can't continue with update"
```
Comment (clacroix12) [inline ocs_ci/utility/kms.py:777]:
```suggestion
            "CSI KMS connection details don't exist, can't continue with update"
```

**Comment by shylesh** [comment]
Verified again after the changes:

./bin/oc get configmap/csi-kms-connection-details -n openshift-storage  --kubeconfig ~/rgeorge/auth/kubeconfig  -o yaml
```yaml
apiVersion: v1
data:
  1-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "rbd-encryption", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "rbd-encryption",
    "VAULT_TOKEN_NAME": "/home/rgeorge/ocs-ci/ocs_ci/templates/CSI/rbd/csi-kms-secret.yaml",
    "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE": "cert.pem", "VAULT_CLIENT_KEY_FILE":
    "privkey.pem"}'
  2-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "rbd-encryption", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "rbd-encryption",
    "VAULT_TOKEN_NAME": "ocs-kms-token", "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE":
    "cert.pem", "VAULT_CLIENT_KEY_FILE": "privkey.pem"}'
  3-vault: '{"KMS_PROVIDER":"vaulttokens","KMS_SERVICE_NAME":"vault","VAULT_ADDR":"https://vault-server:8200","VAULT_BACKEND_PATH":"rbd-encryption","VAULT_CACERT":"ocs-kms-ca-secret-vbi56p","VAULT_TLS_SERVER_NAME":"","VAULT_CLIENT_CERT":"ocs-kms-client-cert-gp13so","VAULT_CLIENT_KEY":"ocs-kms-client-key-3bdpw7","VAULT_NAMESPACE":"ocs/rbd","VAULT_TOKEN_NAME":"ocs-kms-token","VAULT_CACERT_FILE":"fullchain.pem","VAULT_CLIENT_CERT_FILE":"cert.pem","VAULT_CLIENT_KEY_FILE":"privkey.pem"}'
  12-vault: '{"KMS_PROVIDER": "vaulttokens", "KMS_SERVICE_NAME": "vault", "VAULT_ADDR":
    "https://vault-server:8200", "VAULT_BACKEND_PATH": "kv-v2", "VAULT_CACERT":
    "ocs-kms-ca-secret", "VAULT_TLS_SERVER_NAME": "", "VAULT_NAMESPACE": "", "VAULT_TOKEN_NAME":
    "ocs-kms-token", "VAULT_CACERT_FILE": "fullchain.pem", "VAULT_CLIENT_CERT_FILE":
    "cert.pem", "VAULT_CLIENT_KEY_FILE": "privkey.pem"}'
kind: ConfigMap
metadata:
  creationTimestamp: "2021-07-15T08:06:13Z"
  name: csi-kms-connection-details
  namespace: openshift-storage
  resourceVersion: "4292408"
  uid: b027c685-46f1-450a-a76a-eb9580ea595b

File: ocs_ci/utility/kms.py (hunk 3 above)
Comment (shylesh) [reply ocs_ci/utility/kms.py]: done

File: ocs_ci/helpers/helpers.py (hunk 2 above)
Comment (shylesh) [reply ocs_ci/helpers/helpers.py]: done

**Review by clacroix12 — APPROVED** [review]

**Review by vasukulkarni — APPROVED** [review]

**Pull request merged**